src/test/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/PrisonApiClientIntTest.kt

```diff
@@ -85,4 +89,41 @@ class PrisonApiClientIntTest(private val mockPrisonService: MockPrisonService) :
       ),
     )
   }
+
+  @Test
+  fun `can get offender release dates by offenderSentCalcId`() {
+    mockPrisonService.withStub(
+      get("/api/offender-dates/sentence-calculation/-1")
+        .willReturn(
+          aResponse()
+            .withHeader("Content-Type", "application/json")
+            .withStatus(200)
+            .withBody(
+              """{
+                      "dtoPostRecallReleaseDate": "2020-03-22",
+                      "conditionalReleaseDate": "2019-05-24",
+                      "sentenceExpiryDate": "2020-03-24",
+                      "effectiveSentenceEndDate": "2025-05-05",
+                      "tariffExpiredRemovalSchemeEligibilityDate": "2020-06-25",
+                      "approvedParoleDate": "2018-09-27",
+                      "comment": "Some Comment Text",
+                      "reasonCode": "NEW",
+                      "calculatedAt": "2017-08-28T00:00:00"
+                  }""",
+            ),
+        ),
+    )
+    // when
+    val offenderKeyDates = prisonApiClient.getNOMISOffenderKeyDates(-1)
+      .getOrElse { problemMessage -> throw CrdWebException(problemMessage, HttpStatus.NOT_FOUND) }
+    assertThat(offenderKeyDates.calculatedAt).isEqualTo(LocalDateTime.of(2017, 8, 28, 0, 0))
+    assertThat(offenderKeyDates.reasonCode).isEqualTo("NEW")
+    assertThat(offenderKeyDates.comment).isEqualTo("Some Comment Text")
+    assertThat(offenderKeyDates.approvedParoleDate).isEqualTo(LocalDate.of(2018, 9, 27))
+    assertThat(offenderKeyDates.tariffExpiredRemovalSchemeEligibilityDate).isEqualTo(LocalDate.of(2020, 6, 25))
+    assertThat(offenderKeyDates.effectiveSentenceEndDate).isEqualTo(LocalDate.of(2025, 5, 5))
+    assertThat(offenderKeyDates.sentenceExpiryDate).isEqualTo(LocalDate.of(2020, 3, 24))
+    assertThat(offenderKeyDates.conditionalReleaseDate).isEqualTo(LocalDate.of(2019, 5, 24))
+    assertThat(offenderKeyDates.dtoPostRecallReleaseDate).isEqualTo(LocalDate.of(2020, 3, 22))
+  }
```

Comment:
Should add some tests for the error scenarios as well (NOT_FOUND, FORBIDDEN, and the else)

Reply:
can add them here or in unit test or both :)

Reply:
Added parameterised test `get offender release dates by offenderSentCalcId from NOMIS should throw error` in this class to test which covers the requested coverage for error scenarios.